### PR TITLE
Aliases

### PR DESCRIPTION
**Pull request by nulldef**
PR to resolve issue #53

### CODE REVIEW

File: lib/surrealist.rb [hunk 1]
```diff
@@ -26,6 +27,8 @@ module Surrealist
   PARENT_VARIABLE = '@__surrealist_schema_parent'.freeze
```
Comment (nesaulov) [inline lib/surrealist.rb]: I guess we can rename `INSTANCE_VARIABLE` to `SCHEMA_ INSTANCE_VARIABLE`

File: lib/surrealist/builder.rb [hunk 2]
```diff
@@ -8,10 +8,12 @@ class Builder
 
     # @param [Carrier] carrier instance of Surrealist::Carrier
     # @param [Hash] schema the schema defined in the object's class.
+    # @param [Hash] aliases the aliases map defined in the object's class.
     # @param [Object] instance the instance of the object which methods from the schema are called on.
-    def initialize(carrier:, schema:, instance:)
-      @carrier = carrier
-      @schema = schema
+    def initialize(carrier:, schema:, aliases: {}, instance:)
```
Comment (nesaulov) [inline lib/surrealist/builder.rb]: I think `{}` part in `aliases: {}` is redundant, as you already do `instance.class.instance_variable_get(ALIASES_INSTANCE_VARIABLE) || {}` in `Surrealist.find_aliases()`

**Comment by nulldef** [comment]
@nesaulov hi there!
Today I have been found some troubles with realization. I think we should change schema nesting  definition DSL for model to
```
class User
  include Surrealist

  json_schema do
    { nested: NestedClass }
  end
end

class NestedClass do
  include Surrealist

  json_schema do
    { name: String }
  end
end
```
'case we should fetch aliases hash on each nested iteration.
I'm gonna to listen your oppinion about this situation.

**Comment by AlessandroMinali** [comment]
@nulldef for your previous comment, not sure what you mean. Our DSL should already work in that way and we have specs in that exact same form: 

https://github.com/nesaulov/surrealist/blob/master/spec/orms/active_record/models.rb#L293-L308

**Comment by nulldef** [comment]
@AlessandroMinali I kind of agree with you about using `:alias_method` instead of this PR, but @nesaulov suggest that it would be cool feature for this gem. I think we need to have little discussion for it. 
In case of renouncement I can suggest refactoring PR for some gem modules.

**Comment by nesaulov** [comment]
I guess @AlessandroMinali is right on this. This is a handy feature but it comes with a trade-off in performance. @nulldef could you please do as you suggested (refactoring PR for some gem modules) and leave the specs for aliases with usage of `aslias_method` (nested structures in particular) and leave a note in README about using `alias_method` to achieve this behaviour?

**Comment by nulldef** [comment]
@nesaulov Yeah, ok

**Comment by nulldef** [comment]
I'm closing it in favor of #55

**Pull request closed without merging**